project/dexter-vd-cpp/src/java/com/samsung/sec/dexter/vdcpp/util/CDTASTUtil.java

```diff
@@ -0,0 +1,401 @@
+package com.samsung.sec.dexter.vdcpp.util;
+
+import java.math.BigDecimal;
+import java.util.Map;
+
+import org.apache.log4j.Logger;
+import org.eclipse.cdt.core.dom.ast.IASTBinaryExpression;
+import org.eclipse.cdt.core.dom.ast.IASTCastExpression;
+import org.eclipse.cdt.core.dom.ast.IASTDeclSpecifier;
+import org.eclipse.cdt.core.dom.ast.IASTDeclaration;
+import org.eclipse.cdt.core.dom.ast.IASTDeclarator;
+import org.eclipse.cdt.core.dom.ast.IASTEqualsInitializer;
+import org.eclipse.cdt.core.dom.ast.IASTExpression;
+import org.eclipse.cdt.core.dom.ast.IASTFileLocation;
+import org.eclipse.cdt.core.dom.ast.IASTFunctionDefinition;
+import org.eclipse.cdt.core.dom.ast.IASTIdExpression;
+import org.eclipse.cdt.core.dom.ast.IASTLiteralExpression;
+import org.eclipse.cdt.core.dom.ast.IASTName;
+import org.eclipse.cdt.core.dom.ast.IASTNamedTypeSpecifier;
+import org.eclipse.cdt.core.dom.ast.IASTNode;
+import org.eclipse.cdt.core.dom.ast.IASTTranslationUnit;
+import org.eclipse.cdt.core.dom.ast.IBasicType;
+import org.eclipse.cdt.core.dom.ast.IBinding;
+import org.eclipse.cdt.core.dom.ast.IType;
+import org.eclipse.cdt.core.dom.ast.IValue;
+import org.eclipse.cdt.core.dom.ast.IVariable;
+import org.eclipse.cdt.core.dom.ast.IASTStatement;
+import org.eclipse.cdt.core.dom.ast.IASTReturnStatement;
+import org.eclipse.cdt.core.dom.ast.IASTFunctionCallExpression;
+import org.eclipse.cdt.core.dom.ast.IASTPointerOperator;
+import org.eclipse.cdt.core.dom.ast.IPointerType;
+import org.eclipse.cdt.core.dom.ast.IASTUnaryExpression;
+import org.eclipse.cdt.core.dom.ast.ASTTypeUtil;
+
+import com.samsung.sec.dexter.core.analyzer.AnalysisConfig;
+import com.samsung.sec.dexter.core.analyzer.AnalysisResult;
+import com.samsung.sec.dexter.core.checker.Checker;
+import com.samsung.sec.dexter.core.defect.PreOccurence;
+import com.samsung.sec.dexter.core.exception.DexterRuntimeException;
+import com.samsung.sec.dexter.vdcpp.plugin.DexterVdCppPlugin;
+
+public class CDTASTUtil {
+	private final static Logger LOG = Logger.getLogger(DexterVdCppPlugin.class);
+	
+	public static void fillDefectData(final AnalysisConfig config,
```

Comment:
This is recommendation.
It is better to have parameters less than 3.
If you have to use many parameters, it is time to consider Parameter Class
to avoid mistakes and establish readability.